pkg/core/xds/metadata.go

```diff
@@ -150,12 +158,14 @@ func DataplaneMetadataFromXdsMetadata(xdsMetadata *_struct.Struct) *DataplaneMet
 			metadataLog.Error(err, "invalid value in dataplane metadata", "field", fieldDataplaneDataplaneResource, "value", value)
 		}
 		switch r := res.(type) {
-		case *core_mesh.DataplaneResource:
-			metadata.DataplaneResource = r
-		case *core_mesh.ZoneIngressResource:
-			metadata.ZoneIngressResource = r
+		case *core_mesh.DataplaneResource,
+			*core_mesh.ZoneIngressResource:
+			metadata.Resource = r
 		default:
-			metadataLog.Error(err, "invalid value in dataplane metadata", "field", fieldDataplaneDataplaneResource, "value", value)
+			metadataLog.Error(err, "invalid dataplane resource type",
+				"resource", r.GetType(),
+				"field", fieldDataplaneDataplaneResource,
+				"value", value)
```

Comment:
love this formatting of keeping key+value in separate lines. I would do this across the code

Reply:
I bet we could write a Go linter to fix that :)